packages/datetime/src/timePicker.tsx

```diff
@@ -263,28 +263,8 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
     };
 
     private getInputChangeHandler = (unit: TimeUnit) => (e: React.SyntheticEvent<HTMLInputElement>) => {
-        const TWO_DIGITS = /^\d{0,2}$/;
-        const THREE_DIGITS = /^\d{0,3}$/;
         const text = getStringValueFromInputEvent(e);
-
-        let isValid = false;
-        switch (unit) {
-            case TimeUnit.HOUR_24:
-            case TimeUnit.HOUR_12:
-            case TimeUnit.MINUTE:
-            case TimeUnit.SECOND:
-                isValid = TWO_DIGITS.test(text);
-                break;
-            case TimeUnit.MS:
-                isValid = THREE_DIGITS.test(text);
-                break;
-            default:
-                throw Error("Invalid TimeUnit");
-        }
-
-        if (isValid) {
-            this.updateTime(parseInt(text, 10), unit);
-        }
```

Comment:
I think this is not needed since similar validation is done by `isTimeUnitValid()` at later point. I decided to remove it, tests passed and TimePicker start to work correctly.